### DISCUSSION TITLE
Cannot plan or apply envcommon changes using Pipelines v2

### DISCUSSION THREAD

**Discussion by umm0n** [gruntwork-io/knowledge-base#822] [Help]
Hello! You must be getting tired of me by now 😄 

We've been having issues planning or applying changes to envcommon files with the ECS deploy runner, you can check out this [issue](https://github.com/orgs/gruntwork-io/discussions/818) for more context.

I've since migrated from ECS deploy runner to Pipelines v2 but the problem persists, if slightly different. We're not getting any errors now (before I was getting a bunch of them), the pipeline simply hangs up at the step shown below, until it eventually times out.

Observations:
- It doesn't matter how many envcommon files we change. It fails with one or with many.
- Plans and applies to other modules, not envcommon, work as expected.
- There's no error of any sort for me to debug.

I've noticed in the logs below that the path is set to:
```
- Module /home/runner/work/infrastructure-pipelines/infrastructure-pipelines/infrastructure-live/br-staging/sa-east-1/br-staging/services/backend-api
```
But our `infrastructure-live` repo isn't called that, it's called `infra-tf-live`. Could this be causing issues? I'm not aware of any way of changing the name.

Any ideas how to fix this? It's a bit critical, since we're unable to run envcommon changes right now, either via ECS deploy runner or Pipelines v2.

```
time="2024-04-26T08:31:08Z" level=info msg="Executing Terragrunt in working dir br-staging with args: [run-all plan --terragrunt-modules-that-include=../_envcommon/services/ecs-fargate-backend-api.hcl --terragrunt-modules-that-include=../_envcommon/services/ecs-fargate-backend-app-web.hcl --terragrunt-modules-that-include=../_envcommon/services/ecs-fargate-backend-claims.hcl --terragrunt-modules-that-include=../_envcommon/services/ecs-fargate-backend-insurer.hcl --terragrunt-modules-that-include=../_envcommon/services/ecs-fargate-backend-payments.hcl --terragrunt-modules-that-include=../_envcommon/services/ecs-fargate-backend-rewards-points-gen.hcl --terragrunt-modules-that-include=../_envcommon/services/ecs-fargate-backend-sample.hcl]" binary=pipelines version=v0.6.1
time="2024-04-26T08:31:08Z" level=info msg="Filtering out environment variable ACTIONS_ID_TOKEN_REQUEST_TOKEN before executing shell command." binary=pipelines version=v0.6.1
time="2024-04-26T08:31:08Z" level=info msg="Filtering out environment variable ACTIONS_ID_TOKEN_REQUEST_URL before executing shell command." binary=pipelines version=v0.6.1
time=2024-04-26T08:31:27Z level=info msg=The stack at /home/runner/work/infrastructure-pipelines/infrastructure-pipelines/infrastructure-live/br-staging will be processed in the following order for command plan:
Group 1
- Module /home/runner/work/infrastructure-pipelines/infrastructure-pipelines/infrastructure-live/br-staging/sa-east-1/br-staging/services/backend-api
- Module /home/runner/work/infrastructure-pipelines/infrastructure-pipelines/infrastructure-live/br-staging/sa-east-1/br-staging/services/backend-claims
- Module /home/runner/work/infrastructure-pipelines/infrastructure-pipelines/infrastructure-live/br-staging/sa-east-1/br-staging/services/backend-insurer
- Module /home/runner/work/infrastructure-pipelines/infrastructure-pipelines/infrastructure-live/br-staging/sa-east-1/br-staging/services/backend-payments
- Module /home/runner/work/infrastructure-pipelines/infrastructure-pipelines/infrastructure-live/br-staging/sa-east-1/br-staging/services/backend-rewards-points-gen
- Module /home/runner/work/infrastructure-pipelines/infrastructure-pipelines/infrastructure-live/br-staging/sa-east-1/br-staging/services/backend-sample
- Module /home/runner/work/infrastructure-pipelines/infrastructure-pipelines/infrastructure-live/br-staging/sa-east-1/br-staging/services/backend-web-app
```

---

EDIT: we're already running the pipelines in debug mode, by setting `--log-level debug` in `deploy-infra.sh`

<ins datetime="2024-04-26T10:29:11Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110908">Tracked in ticket #110908</a></p>
</ins>

**Comment by josh-padnick** [comment]
Not a problem! We'll discuss internally and get back to you shortly.

**Comment by josh-padnick** [comment]
Hi @umm0n, we noticed that you've got a few tickets open all around the same topic and we're now setting up customers on our latest version of Pipelines -- you can see a preview at https://github.com/gruntwork-io/docs/pull/1683. I'll email you directly now to see if we can get a time to get you onboarded to that.

Once you're on that, you can work directly with our engineering team to address any types of issues above, and we'll have a mechanism to quickly push updates to you if needed.
